clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -52,6 +59,118 @@ abstract class OpenstackClientProvider {
+   * @param lbPoolId
+   * @param instanceId
+   */
+  void registerInstancesWithLoadBalancer(String lbPoolId, String instanceId) {
+
+    //weight of member
+    int weight = 1
+
+    //get pool
+    LbPool pool = getPool(lbPoolId)
+
+    //port from load balancer pool
+    int internalPort = getInternalLoadBalancerPort(pool)
+
+    //lookup instance ip
+    String ip = getIpForInstance(instanceId)
+
+    //add to pool
+    //TODO use handleRequest once that is refactored
+    client.networking().loadbalancers().member().create(
+      Builders.member().address(ip).poolId(lbPoolId).protocolPort(internalPort).weight(weight).build()
+    )
+  }
+
+  /**
+   * Unassociate an instance with a load balancer.
+   * @param lbPoolId
+   * @param instanceId
+   */
+  void deregisterInstancesFromLoadBalancer(String lbPoolId, String instanceId) {
+
+    //lookup instance ip
+    String ip = getIpForInstance(instanceId)
+
+    //get pool
+    LbPool pool = getPool(lbPoolId)
+
+    //get member with matching IP
+    //TODO use handleRequest once that is refactored
+    String memberId = client.networking().loadbalancers().member().list().find { m -> m.address == ip }?.id
+    if (StringUtils.isEmpty(memberId)) {
+      throw new OpenstackOperationException("Instance=${instanceId} with ip=${ip} is not associated with any load balancer memberships")
+    }
+    if (pool.members.find { it == memberId} == null) {
+      throw new OpenstackOperationException("Member id=${memberId} is not associated with load balancer with id=${lbPoolId}")
+    }
+
+    //remove from pool
+    //TODO use handleRequest once that is refactored
+    client.networking().loadbalancers().member().delete(memberId)
+  }
+
+  /**
+   * Get port from load balancer description. Openstack load balancers have no native concept of internal port,
+   * so we store in the description field of the load balancer.
+   * this may be changed in a future version.
+   * @param pool
+   * @return
+   */
+  int getInternalLoadBalancerPort(LbPool pool) {
+    int minPort = 1
+    int maxPort = (1 << 16) - 1
+    String descriptionPattern = ".*internal_port=([0-9]+).*"
+
+    Pattern pattern = Pattern.compile(descriptionPattern)
+    Matcher matcher = pattern.matcher(pool.description)
+    int internalPort = 0
+    if (matcher.matches()) {
+      internalPort = matcher.group(1).toInteger()
+    }
+    if (internalPort < minPort || internalPort > maxPort) {
+      throw new OpenstackOperationException("Internal pool port is outside of the valid range.")
+    }
+    internalPort
+  }
+
+  /**
+   * Get an IP address from a server.
+   * @param server
+   * @return
+   */
+  String getIpForInstance(String instanceId) {
+    //TODO use handleRequest once that is refactored
+    Server server = client.compute().servers().get(instanceId)
+    if (server == null) {
+      throw new OpenstackOperationException("Could not find server with id=${instanceId}")
+    }
+    //just get the first address found
```

Comment:
It depends on how we are creating instances. If we know the network name, that would help. But there can even be multiple ip addresses for a specific network. Do we pick ones that start with 192.*?